File: tests/rustdoc-js/macro-kinds.js
// exact-check

const EXPECTED = [
    {
        'query': 'macro:macro',
        'others': [
            {
                'path': 'macro_kinds',
                'name': 'macro1',
                'href': '../macro_kinds/macro.macro1.html',
                'ty': 16,
            },
            {
                'path': 'macro_kinds',
                'name': 'macro1',
                'href': '../macro_kinds/macro.macro1.html',
                'ty': 23,
            },
            {
                'path': 'macro_kinds',
                'name': 'macro1',
                'href': '../macro_kinds/macro.macro1.html',
                'ty': 24,
            },
            {
                'path': 'macro_kinds',
                'name': 'macro2',
                'href': '../macro_kinds/attr.macro2.html',
                'ty': 23,
            },
            {
                'path': 'macro_kinds',
                'name': 'macro4',
                'href': '../macro_kinds/derive.macro4.html',
                'ty': 24,
            },
            {
                'path': 'macro_kinds',
                'name': 'macro3',
                'href': '../macro_kinds/macro.macro3.html',
                'ty': 16,
            },
        ],
    },
    {
        'query': 'attr:macro',
        'others': [
            { 'path': 'macro_kinds', 'name': 'macro1', 'href': '../macro_kinds/macro.macro1.html' },
            { 'path': 'macro_kinds', 'name': 'macro2', 'href': '../macro_kinds/attr.macro2.html' },
        ],
    },
    {
        'query': 'derive:macro',
        'others': [
            { 'path': 'macro_kinds', 'name': 'macro1', 'href': '../macro_kinds/macro.macro1.html' },
            {
                'path': 'macro_kinds', 'name': 'macro4', 'href': '../macro_kinds/derive.macro4.html'
            },
        ],
    },
];

File: tests/rustdoc-js/macro-kinds.rs
// Test which ensures that attribute macros are correctly handled by the search.
// For example: `macro1` should appear in both `attr` and `macro` filters whereas
// `macro2` and `macro3` should only appear in `attr` or `macro` filters respectively.

#![feature(macro_attr)]
#![feature(macro_derive)]

#[macro_export]
macro_rules! macro1 {
    attr() () => {};
    derive() () => {};
    () => {};
}

#[macro_export]
macro_rules! macro2 {
    attr() () => {};
}

#[macro_export]
macro_rules! macro3 {
    () => {};
}

#[macro_export]
macro_rules! macro4 {
    derive() () => {};
}

#[macro_export]
macro_rules! macro5 {
    attr() () => {};
    derive() () => {};
}
